spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -275,6 +287,16 @@ contract PerpetualTranche is
         _;
     }
 
+    /// @dev Throws if called by any account other than an authorized roller.
+    modifier onlyAuthorizedRollers() {
+        // If the set it empty, permit all callers
+        // else permit only whitelisted callers.
+        if (_authorizedRollers.length() > 0 && !_authorizedRollers.contains(_msgSender())) {
+            revert UnauthorizedCall(_msgSender(), address(0));
```

Comment:
`UnauthorizedCall` is triggered when a non roller invokes rollover. 

`RollerAuthorizationFailed` is triggered when the owner tries to authorize a new roller which fails because the address is already authorized